syncstorage/tests/functional/test_storage.py

```diff
@@ -1378,6 +1378,140 @@ def test_we_dont_need_no_stinkin_batches(self):
+        orig_ts = float(resp.headers['X-Last-Modified'])
+
+        # Update one, and add a new one.
+        bsos = [
+          {'id': 'b', 'payload': 'bii'},
+          {'id': 'c', 'payload': 'sea'},
+        ]
+        resp = self.app.post_json(collection + '?batch=true', bsos)
+        batch = resp.json["batch"]
+
+        # The updated item hasn't been written yet.
+        resp = self.app.get(collection + '?full=1')
+        res = resp.json
+        res.sort(key=lambda bso: bso['id'])
+        self.assertEquals(len(res), 2)
+        self.assertEquals(res[0]['payload'], 'aai')
+        self.assertEquals(res[1]['payload'], 'bee')
+        self.assertEquals(res[0]['modified'], orig_ts)
+        self.assertEquals(res[1]['modified'], orig_ts)
+        self.assertEquals(res[1]['sortindex'], 17)
+
+        endpoint = collection + '?batch={0}&commit=true'.format(batch)
+        resp = self.app.post_json(endpoint, [])
+        commit_ts = float(resp.headers['X-Last-Modified'])
+
+        # The changes have now been applied.
+        resp = self.app.get(collection + '?full=1')
+        res = resp.json
+        res.sort(key=lambda bso: bso['id'])
+        self.assertEquals(len(res), 3)
+        self.assertEquals(res[0]['payload'], 'aai')
+        self.assertEquals(res[1]['payload'], 'bii')
+        self.assertEquals(res[2]['payload'], 'sea')
+        self.assertEquals(res[0]['modified'], orig_ts)
+        self.assertEquals(res[1]['modified'], commit_ts)
+        self.assertEquals(res[2]['modified'], commit_ts)
+
+        # Fields not touched by the batch, should have been preserved.
+        self.assertEquals(res[1]['sortindex'], 17)
+
+    def test_batch_ttl_update(self):
+        collection = self.root + '/storage/col2'
+        bsos = [
+          {'id': 'a', 'payload': 'ayy'},
+          {'id': 'b', 'payload': 'bea'},
+          {'id': 'c', 'payload': 'see'}
+        ]
+        resp = self.app.post_json(collection, bsos)
+
+        # Bump ttls as a series of individual batch operations.
+        resp = self.app.post_json(collection + '?batch=true', [], status=202)
+        batch = resp.json["batch"]
+        endpoint = collection + '?batch={0}'.format(batch)
+        resp = self.app.post_json(endpoint, [{'id': 'a', 'ttl': 2}],
+                                  status=202)
+        resp = self.app.post_json(endpoint, [{'id': 'b', 'ttl': 2}],
+                                  status=202)
+        resp = self.app.post_json(endpoint + '&commit=true', [],
+                                  status=200)
+
+        # The payloads should be unchanged
+        resp = self.app.get(collection + '?full=1')
+        res = resp.json
+        res.sort(key=lambda bso: bso['id'])
+        self.assertEquals(len(res), 3)
+        self.assertEquals(res[0]['payload'], 'ayy')
+        self.assertEquals(res[1]['payload'], 'bea')
+        self.assertEquals(res[2]['payload'], 'see')
+
+        # If we wait, the ttls should kick in
```

Comment:
Unresolved semantic issue: should the ttl time count from the time at which the item is posted, or the time at which the batch is committed?  It probably doesn't matter, but perhaps we should pick one and be explicit about it.